### ISSUE TITLE
FR: Enable browse by category in the Palette Manager

### ISSUE THREAD

**Issue by knolleary** [node-red/node-red#4933]
We're introducing categories to the public node catalogue. This will be reflected in the catalog json the editor loads as a `categories` property which is an array of string category names. This isn't quite live yet - will update this issue when its done.

We want to update the Manage Palette to allow the user to browse/filter the catalog by these categories. It's also an chance to reexamine the overall UX of the palette manager when browsing. Want to look at making it easier to find useful nodes.

**Comment by GogoVega** [comment]
For the filter input, do you plan to do it like the search bar? (a drop-down menu)

<img width="476" alt="Capture d’écran 2024-11-11 à 13 26 18" src="https://github.com/user-attachments/assets/994e2fb3-b5af-4b6c-be4f-3f3c9ff0540a">

The fastest would be to add a multiple choice input via TypedInput but it takes up space. It could be hidden and appear with a button icon.

For the list of categories, it might be better to load it from the catalog so as not to hard code it in the palette.

**Comment by knolleary** [comment]
> For the list of categories, it might be better to load it from the catalog so as not to hard code it in the palette.

Yes - the list of categories available to filter by will be determined by whatever categories are present in the catalog.

I don't have a UI design yet, we already have sort options in the header. Want to find a nice way to include filter options that isn't too hidden. I have a couple half-ideas for it, but will want to experiement.

<img width="592" alt="image" src="https://github.com/user-attachments/assets/ba176a98-21a9-4536-a59b-e3873e81ec43">
